core/src/main/java/com/linecorp/armeria/internal/common/stream/SurroundingPublisher.java

```diff
@@ -67,6 +75,22 @@ public SurroundingPublisher(@Nullable T head, Publisher<? extends T> publisher,
             this.publisher = new PublisherBasedStreamMessage<>(publisher);
         }
         this.tail = tail;
+        finalizer = null;
```

Comment:
Should we convert `tail` to `finalizer` and delegate instead?
```java
static HttpResponse of(ResponseHeaders headers, 
                       Publisher<? extends HttpData> publisher, 
                       HttpHeaders trailers) {
    return of(headers, publisher, ignored -> trailers);
}

static HttpResponse of(ResponseHeaders headers, 
                       Publisher<? extends HttpData> publisher, 
                       Function<@Nullable Throwable, HttpHeaders> trailersFunction) { ... }
```

Reply:
oh it's nice! ✅ updated.

If we use `tailersFunction: ignored -> trailers` on `HttpResponse.of`, `trailers` is emitted when publisher error. is it okay?